dkg/pedersen/handler.go

```diff
@@ -123,8 +167,11 @@ func (h *Handler) Stream(out mino.Sender, in mino.Receiver) error {
 	}
 	h.Unlock()
 
-	deals := list.New()
-	responses := list.New()
+	deals := newCryChan[types.Deal](chanSize)
+	responses := newCryChan[*pedersen.Response](chanSize)
+
+	globalCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
we'll know about it, but this will still block indefinitely right?
there's no timeout mechanism in place, shouldn't we keep one?

especially when we run this with lots of nodes, it's hard to tell if we're stuck or still doing actual useful work.

Reply:
It blocks until the function exits. This context's job is to make sure we don't have a stale go routine in case the function exits badly. Having a timeout on the receiving loop is another question. It should be the orchestrator's responsibility (the one that starts the setup) to terminate the stream after some timeout, thus making the function exit.